src/GNDStk/gnds-keyword.hpp

```diff
@@ -20,97 +20,114 @@ class children {
 // meta_t
 // child_t
 // Forward declarations, which we'll need for other constructs in this file.
+// fixme See if I can make *forward* declarations unnecessary.
 // -----------------------------------------------------------------------------
 
 // meta_t
 // The default template parameter, std::string, means to retrieve the metadatum
 // in its original form in the tree: as a std::string.
-template<class T = std::string>
+template<
+   class RESULT = std::string
+>
 class meta_t;
 
 // child_t
 // The default template parameter, void, means to retrieve the child node
 // in its original form in the tree: as some variation of a Node<>. The precise
 // type thus received depends on the template parameters of the tree's Tree<>
-// type, as those parameters trickle down into the Tree<>'s Node<>s. The fact
-// that there's no fixed "node" type is why we use void for this meaning.
-template<class T = void, class METADATA=metadata<>, class CHILDREN=children<>>
+// type, because those parameters trickle down into the Tree<>'s Node<>s. The
+// fact that there's no fixed Node<> type is why we use void for this meaning.
+// See class child_t's definition, elsewhere, for an in-depth description.
+template<
+   class RESULT   = void,
+   bool  MULTIPLE = false,      // multiple appearances allowed (e.g. <axis>)
+   class METADATA = metadata<>, // allowed metadata for this child-node type
+   class CHILDREN = children<>  // allowed children ...
+>
 class child_t;
 
 
 
 // -----------------------------------------------------------------------------
-// mcat
-// ccat
+// catenateMetadata
+// catenateChildren
+// fixme Eventually, I should put in a description, probably via an example,
+// of what these template metaprogramming constructs are all about.
 // -----------------------------------------------------------------------------
 
 namespace detail {
 
 // ------------------------
-// mcat
+// catenateMetadata
 // metadata<> += meta_ts
 // chuck child_ts
 // ------------------------
 
 // default
-template<class METADATA, class... Ks> class mcat { };
+template<class METADATA, class... Ks> class catenateMetadata { };
```

Comment:
I'm not sure what all the `catenate...` classes are for. Let's talk about this.

Reply:
This is all some template metaprogramming machinery that's intended to serve as a foundation for a "keyword building" capability that I haven't yet entirely finished. Basically, they'll make it much easier for users to create amazing and fabulous capabilities, without having to think too hard about certain details that my metaprogramming stuff can do for them.